products/mlengine/api.yaml

```diff
@@ -235,4 +235,338 @@ objects:
           implements the Predictor interface described in this reference
           field. The module containing this class should be included in a
           package provided to the packageUris field.
-
+  - !ruby/object:Api::Resource
+    name: 'Job'
+    base_url: projects/{{project}}/jobs
+    self_link: projects/{{project}}/jobs/{{name}}
+    input: true
+    description: |
+      Represents a training or prediction job.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'jobId'
+        description: Required. The user-specified id of the job.
+        required: true
+      - !ruby/object:Api::Type::Time
+        name: 'createTime'
+        description: When the job was created.
+        output: true
+      - !ruby/object:Api::Type::Time
+        name: 'endTime'
+        description: When the job processing was completed.
+        output: true     
+      - !ruby/object:Api::Type::Enum
+        name: 'state'
+        description: The detailed state of a job.
+        values:
+          - STATE_UNSPECIFIED
+          - QUEUED
+          - PREPARING
+          - RUNNING
+          - SUCCEEDED
+          - FAILED
+          - CANCELLING
+          - CANCELLED
+        output: true     
+      - !ruby/object:Api::Type::String
+        name: 'errorMessage'
+        description: The details of a failure or a cancellation.
+        output: true                
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: |
+           One or more labels that you can add, to organize your jobs. Each label is a key-value pair, 
+           where both the key and the value are arbitrary Strings that you supply. 
+           An object containing a list of "key": value pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+      # no etag since it's input: true     
+      ## trainingInput starts here     
+      - !ruby/object:Api::Type::NestedObject
+        name: 'trainingInput'
+        description: Input parameters to create a training job.
+        output: false
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'scaleTier'
+            description: Specifies the machine types, the number of replicas for workers and parameter servers.
+            values:
+              - BASIC
+              - STANDARD_1
+              - PREMIUM_1
+              - BASIC_GPU
+              - BASIC_TPU
+              - CUSTOM
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'masterType'
+            description: Specifies the type of virtual machine to use for your training job's master worker  
```

Comment:
Can you also add the guidance of "You must specify this field when scaleTier is set to CUSTOM."?